Neos.Fusion/Classes/Core/RuntimeFactory.php

```diff
@@ -39,24 +45,38 @@ public function create(array $fusionConfiguration, ControllerContext $controller
         if ($controllerContext === null) {
             $controllerContext = self::createControllerContextFromEnvironment();
         }
-        return new Runtime(
+        $defaultContextVariables = EelUtility::getDefaultContextVariables(
+            $this->defaultContextConfiguration ?? []
+        );
+        $runtime = new Runtime(
             FusionConfiguration::fromArray($fusionConfiguration),
-            $controllerContext
+            FusionDefaultContextVariables::fromArray(
+                ['request' => $controllerContext->getRequest(), ...$defaultContextVariables]
+            )
         );
+        $runtime->setControllerContext($controllerContext);
+        return $runtime;
     }
 
-    public function createFromConfiguration(FusionConfiguration $fusionConfiguration, ControllerContext $controllerContext): Runtime
-    {
-        return new Runtime($fusionConfiguration, $controllerContext);
+    public function createFromConfiguration(
+        FusionConfiguration $fusionConfiguration,
+        FusionDefaultContextVariables $additionalDefaultContextVariables
+    ): Runtime {
+        $defaultContextVariables = FusionDefaultContextVariables::fromArray(
+            EelUtility::getDefaultContextVariables(
+                $this->defaultContextConfiguration ?? []
+            )
+        );
+        return new Runtime($fusionConfiguration, $defaultContextVariables->merge($additionalDefaultContextVariables));
```

Comment:
im currently enriching the `$additionalDefaultContextVariables` with the eel helpers from the config. instead of using two kinds of different dto classes (one only for the factory) - i think this makes the most sense?

Reply:
yes after the rename to `FusionGlobals` it makes totally sense!